### ISSUE TITLE
How to connect to an onion daemon using the public available proxy?

### ISSUE THREAD

**Issue by jjmmbb** [tdex-network/tdex-app#601]
iOS version is not connecting to onion daemon even if you change configuration to use internal Tor Proxy. Still https://proxy.tdex.network working?

**Comment by tiero** [comment]
Proxy only works if the onion daemon is in the registry. Have you opened PR yet?

**Comment by jjmmbb** [comment]
Only registered on registry can use the app? But what if you manually type the server address on the list?

**Comment by tiero** [comment]
If the server is only reachable on the onion address, you need the app to either a) use Orbot b) embed Tor inside (we tried but was clunky)

A solution is having proxy ran by the community (or ideally TDEX providers SHOULD do that!) which we implemented. To avoid using our proxy as free for all "tor to web" proxy which can cause trouble, we decided to have the inclusion in the registry as a way to "whitelist" the onion address that the proxy that is run PRO-BONO by other tdex developers.

You can (and should) run your own proxy, simply passing your onion address to start, no need for the public registry.
https://github.com/tdex-network/tor-proxy

```
$ torproxy start --insecure --registry '[{"endpoint": "http://somewherefaraway.onion:80"}]' 
```

**Comment by jjmmbb** [comment]
Apparently the app can not connect to any onion servers. I am trying all registered servers using proxy.tdex.network and it's not working. Only one token is appearing and probable from the clearnet server. I also have tried the proxy using all onion addresses in the app list as single endpoint using port 7070 and no one connect. It's always asking to refresh the list if I keep just one onion server to test. I have tried that USDT server in the list and nothing appears.

**Comment by jjmmbb** [comment]
Somebody?

**Comment by tiero** [comment]
Have you tried running you own proxy locally and connect the app to it? It's one command and you can specify your own list of providers.

**Comment by jjmmbb** [comment]
Yes, I have tried it:

torproxy start --insecure --registry '[{"endpoint": "http://somewherefaraway.onion:80"}]'

**Comment by tiero** [comment]
I assume you have swtitched here?

![image](https://github.com/user-attachments/assets/98bd8797-f19f-42cf-bfd9-be57998e577f)

**Comment by jjmmbb** [comment]
Yes! I have. You can test. Keep the https://proxy.tdex.network on that configuration and remove servers from providers list and keep one server from ONION. When you refresh the app it says that there is no providers and force you to refresh.

**Comment by jjmmbb** [comment]
I have tested every single onion host in the registry. I just delete everything in the provider list, and then I've added one by one. When click in exchange icon I just get a message asking to refresh the list from json file and all hosts are restored.

**Comment by tiero** [comment]
Thank you for testing! I hope some contributor will have time to investigate and make it more robust the code in the app! Pretty tight up lately on my side

**Comment by jjmmbb** [comment]
Have you tried the way I have described? Do you have the same issue?

**Comment by jjmmbb** [comment]
@tiero I really think that the proxy from tdex.network are not working. I am really unable to connect to providers using tdex app. Is this app still working for you?

**Comment by tiero** [comment]
I can curl Onions providers from the terminal so the proxy seems to work as expected. (assuming the onion providers are online, sometimes they are offline or tor network itself is down) 

The proxy is very simple piece of software to run, ideally always best to self-host your own, but has always been online AFAICT.

Also the apps is not super sophisticated with retries and whatnot. The app it's more of a reference implementation, than a commercial product. Contributions are welcome.

**Comment by jjmmbb** [comment]
Hello @tiero,

On my side, there are all those servers:

`[
  {
    "name": "Tdex_DEPIX",
    "endpoint": "http://dyki2gt6yno6g32hzpsclxtqhdl4jyebl36oiu2u2r5wgqwzmykigoqd.onion:80"
  },
  {
    "name": "Tdex_LCAD",
    "endpoint": "http://d7y3mzol3eo2tneqw5oytj23knm3734npwml4jzazrzzpy32e56lrxqd.onion:80"
  },
  {
    "name": "ONLYBESTOFFERS.TRADE",
    "endpoint": "http://onlybestyqgaaloz2azrefzu6kkrtakrh324ana2fqfrzpves2rgsqyd.onion:9945"
  },
  {
    "name": "eldamar",
    "endpoint": "https://tdex.eldamar.icu:443"
  },
  {
    "name": "Tdex_USDT",
    "endpoint": "http://btuiucyiumvdzhlnhqbnj6k5yrbwo2dlznzosicwnf6nw5aewr77elyd.onion:80"
  }
]`

And one more not present on this list. Let's use the first one as example. Via ONION I can access the endpoint. Aparrently only the ONLYBESTOFFERS are not working. So as the host is accessible via ONION, I removed all other servers and just keep the first one and when I click in the swap botton the app says there is no providers available and just refresh the list and there are only two coins available. The same happens to you?



> I can curl Onions providers from the terminal so the proxy seems to work as expected. (assuming the onion providers are online, sometimes they are offline or tor network itself is down)
> 
> The proxy is very simple piece of software to run, ideally always best to self-host your own, but has always been online AFAICT.
> 
> Also the apps is not super sophisticated with retries and whatnot. The app it's more of a reference implementation, than a commercial product. Contributions are welcome.

**Comment by jjmmbb** [comment]
@tiero I have tried all possibilities even using https://proxy.tdex.network.

Same error using my own cloned tdex-torproxy. Just one observation: endpoints are http://TORHOST.ONION:80

If you access: https://proxy.tdex.network/TORHOST (without dot onion, without port and without protocol), I can access available servers. Maybe the iOS version of app is passing the endpoint full address to proxy without any filter. (just guessing).

I have also checked TDEX-EXPLORER and there are only two servers that is really running (Tdex_DEPIX and Tdex_CAD), but always is displayed the error below:

![Image](https://github.com/user-attachments/assets/b8fac7bf-915d-404a-b2f6-d05d33561483)
